Makefile

```diff
@@ -1,59 +1,61 @@
+ifneq ($(origin CHANNELS), undefined)
+BUNDLE_CHANNELS := --channels=$(CHANNELS)
+endif
+
+# DEFAULT_CHANNEL defines the default channel used in the bundle.
+# Add a new line here if you would like to change its default config. (E.g DEFAULT_CHANNEL = "stable")
+# To re-generate a bundle for any other default channel without changing the default setup, you can:
+# - use the DEFAULT_CHANNEL as arg of the bundle target (e.g make bundle DEFAULT_CHANNEL=stable)
+# - use environment variables to overwrite this value (e.g export DEFAULT_CHANNEL="stable")
+ifneq ($(origin DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# IMAGE_TAG_BASE defines the docker.io namespace and part of the image name for remote images.
+# This variable is used to construct full image tags for bundle and catalog images.
 #
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Package related
-BINARY_NAME=network-operator
-PACKAGE=network-operator
-ORG_PATH=github.com/Mellanox
-REPO_PATH=$(ORG_PATH)/$(PACKAGE)
-CHART_PATH=$(CURDIR)/deployment/$(PACKAGE)
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# mellanox.com/network-operator-bundle:$VERSION and mellanox.com/network-operator-catalog:$VERSION.
+IMAGE_TAG_BASE ?= mellanox.com/network-operator
+
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= docker.io/mellanox/network-operator:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Other vars
 GOPATH?=$(CURDIR)/.gopath
-GOBIN =$(CURDIR)/bin
 TOOLSDIR=$(CURDIR)/bin
-BUILDDIR=$(CURDIR)/build/_output
+
+ORG_PATH=github.com/Mellanox
+PACKAGE=network-operator
+REPO_PATH=$(ORG_PATH)/$(PACKAGE)
 BASE=$(GOPATH)/src/$(REPO_PATH)
-GOFILES=$(shell find . -name "*.go" | grep -vE "(\/vendor\/)|(_test.go)")
-PKGS=$(or $(PKG),$(shell cd $(BASE) && env GOPATH=$(GOPATH) $(GO) list ./... | grep -v "^$(PACKAGE)/vendor/"))
-TESTPKGS = $(shell env GOPATH=$(GOPATH) $(GO) list -f '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' $(PKGS))
-
-export GOPATH
-export GOBIN
```

Comment:
these two would need to be exported in the modified Makefile as well.

also GO<PATH|BIN> initialization logic is different in the modified Makefile.

Reply:
I used the new initialization logic, why would we want to preserve the old?